### PR TITLE
Error log message that was missing

### PR DESCRIPTION
**Pull request by majikandy**
If the queue doesn't exist then we don't get an error message when a request comes in.

### Acceptance Criteria

- [ ] Error in the logs when queue not found

### Testing information
Point at a non-existent queue and try a delete or create

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: medicines/doc-index-updater/src/document_manager/mod.rs [hunk 1]
```diff
@@ -54,6 +54,7 @@ where
                     },
                 )
                 .await?;
+            tracing::error!("Failed to dispatch to queue. Check API keys");
```
Comment (rjkerrison) [inline medicines/doc-index-updater/src/document_manager/mod.rs:57]: can you put this above the state_manager call? a failure in state_manager will prevent us from hitting this line. You could also use `Err(e)` and print the true error message of `e`.

**Comment by majikandy** [comment]
@craiga @TimboTambo @rjkerrison 

Added in the instrumentation to get correlation id in it too

log message now looks like this:-

```
ERROR document_manager{correlation_id="3c9ab764-4f84-4f45-85ac-a9a6678d17cf"}: doc_index_updater::document_manager: UnexpectedHTTPResult(UnexpectedHTTPResult { expected: [201], received: 401, body: "" }). Failed to dispatch to queue. Check API Keys or Policy values in environment variables.
```

File: medicines/doc-index-updater/src/document_manager/mod.rs [hunk 2]
```diff
@@ -67,12 +72,16 @@ async fn delete_document_handler(
...
+        let correlation_id = id.to_string();
+        let correlation_id = correlation_id.as_str();
 
         let message = DeleteMessage {
             job_id: id,
             document_content_id,
         };
-        queue_job(&mut queue, &state_manager, message).await
+        queue_job(&mut queue, &state_manager, message)
+            .instrument(tracing::info_span!("document_manager", correlation_id))
```
Comment (craiga) [inline medicines/doc-index-updater/src/document_manager/mod.rs:83]: :+1:

Would be nice to do the same in create document handler.

Comment (majikandy) [reply medicines/doc-index-updater/src/document_manager/mod.rs:84]: true, this is like never ending story :)

Comment (majikandy) [reply medicines/doc-index-updater/src/document_manager/mod.rs]: Done

**Pull request merged**